Sources/protoc-gen-swift/FieldGenerator.swift

```diff
@@ -199,6 +187,28 @@ class FieldGeneratorBase {
         }
     }
 
+    /// Generates the Swift expression that will be used by the field's accessors to specify the
+    /// field's offset in memory, taking into account the target platform.
+    var storageOffsetExpression: String {
+        // If all the values are the same, generate cleaner code by just passing the single value
+        // directly.
+        if let valueIfAllEqual = storageOffsets.valueIfAllEqual {
+            return "\(valueIfAllEqual)"
+        }
+
+        // Otherwise, generate a call to the helper function that chooses the right value based on
+        // target platform.
+        var result = "SwiftProtobuf._fieldOffset("
```

Comment:
Aside - I wonder if `reduce` would be a good pattern for building up strings like this, see the initial value, and then just keep adding, etc. You could even see with `""`,  and use that input as the signal for the initial block vs. the `", "` to add.